llvm/lib/Transforms/IPO/MemProfContextDisambiguation.cpp

```diff
@@ -2468,8 +2533,10 @@ void CallsiteContextGraph<DerivedCCG, FuncTy, CallTy>::
 template <typename DerivedCCG, typename FuncTy, typename CallTy>
 void CallsiteContextGraph<DerivedCCG, FuncTy, CallTy>::identifyClones() {
   DenseSet<const ContextNode *> Visited;
-  for (auto &Entry : AllocationCallToContextNodeMap)
-    identifyClones(Entry.second, Visited);
+  for (auto &Entry : AllocationCallToContextNodeMap) {
+    Visited.clear();
```

Comment:
Was this `clear()` not necessary before?

Reply:
No it was not necessary, because before we were attempting to do all cloning for each node once. Now we may revisit each node to consider cloning for each allocation it affects separately.